photon-ml/src/test/scala/com/linkedin/photon/ml/optimization/LBFGSTest.scala

```diff
@@ -16,115 +16,66 @@ package com.linkedin.photon.ml.optimization
 
-
 /**
- * Test that LBFGS/OWLQN can shrink the coefficients to zero.
- *
- * The objective function is (x-1)^2 + (y-4)^2, with L1 regularization C(|x| + |y|). We are only interested in the behavior around
- * the minimum, where x > 0 and y > 0. Thus the function to be optimized then become (x-1)^2 + (y-4)^2 + C(x+y) which has obvious analytic
- * solution. The test is based on this function and verifies the shrinkage of x and y.
- *
- * @author dpeng
- * @author nkatariy
- */
-
+  * Test that LBFGS can shrink the coefficients to zero.
+  *
+  * The objective function is (x-1)^2 + (y-4)^2. We are only interested in the behavior around the minimum, where x > 0
+  * and y > 0, which has obvious analytic solution. The test is based on this function and verifies the shrinkage of x
+  * and y.
+  */
 class LBFGSTest {
-  val epsilon = 1.0E-6
-  val random = new Random(1)
+  val random = new Random(LBFGSTest.RANDOM_SEED)
 
   private def getRandomInput(dim: Int): DenseVector[Double] = {
     DenseVector(Seq.fill(dim)(random.nextGaussian).toArray)
   }
 
-  @Test(dataProvider = "dataProvider")
-  def testLBFGS(l1Weight: Double, constraintMap: Option[Map[Int, (Double, Double)]],
-                minimum: Array[Double], expectedValue: Double): Unit = {
-    val lbfgs = new LBFGS[LabeledPoint]
-    lbfgs.setConstraintMap(constraintMap)
-
-    var testFunc: DiffFunction[LabeledPoint] = null
-    var dummyData: Array[LabeledPoint] = null
-    var initCoeff: DenseVector[Double] = null
-
-    constraintMap match {
-      case Some(x) =>
-        testFunc = if (l1Weight == 0) {
-          new Test1DFunction
-        } else {
-          DiffFunction.withRegularization(new Test1DFunction, L1RegularizationContext, l1Weight)
-        }
-        dummyData = Array(LabeledPoint(0.0, DenseVector[Double](0.0), 0.0, 0.0))
-        initCoeff = getRandomInput(1)
-      case None =>
-        testFunc = if (l1Weight == 0) {
-          new Test2DFunction
-        } else {
-          DiffFunction.withRegularization(new Test2DFunction, L1RegularizationContext, l1Weight)
-        }
-        dummyData = Array(LabeledPoint(0.0, DenseVector[Double](0.0, 0.0), 0.0, 0.0))
-        initCoeff = getRandomInput(2)
-    }
-    val (coef, actualValue) = lbfgs.optimize(dummyData, testFunc, initCoeff)
-    Assertions.assertIterableEqualsWithTolerance(coef.toArray, minimum, epsilon)
-    Assert.assertEquals(actualValue, expectedValue, epsilon)
-  }
-
   @DataProvider(name = "dataProvider")
   def dataProvider(): Array[Array[Any]] = {
     Array(
-      Array(0.0, None, Array(1.0, 4.0), 0.0),
-      Array(1.0, None, Array(0.5, 3.5), 4.5),
-      Array(2.0, None, Array(0.0, 3.0), 8.0),
-      Array(8.0, None, Array(0.0, 0.0), 17.0),
+      Array(None, Array(4.0, 1.0)),
 
-      // note that expected value here is the value of the function at the unconstrained optima since the
+      // Note that expected value here is the value of the function at the unconstrained optima since the
```

Comment:
@joshvfleming 
Removed regularization weight (that's tested for OWLQN now) and expected value (it should be static now that the regularization weight is fixed at 0). Played around with the test values to be clearer that we're checking that they're being bounded correctly.